src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
-export interface Figure {
+type ShapeOfFigure = 'triangle' | 'circle' | 'rectangle';
+type ColorOfFigure = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: ShapeOfFigure;
+  color: ColorOfFigure;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: ShapeOfFigure = 'triangle';
 
-}
+  constructor(
+    public color: ColorOfFigure,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (Math.min(this.a, this.b, this.c) <= 0
+    || [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[2]
+    >= [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[0]
+    + [this.a, this.b, this.c].sort((prev, cur) => prev - cur)[1]) {
```

Comment:
🤯

Reply:
Really hard to read. You can create two separate errors: if you have a side smaller than 0; if the longest side of a triangle is >= than a sum of two others.